packages/modules/src/document/Document.ts

```diff
@@ -370,6 +370,59 @@ export async function fromContent({
   verifyDataStructure(document)
   return document
 }
+export async function updateStream(
+  document: IDocument,
```

Comment:
Isn't it a better approach to pass the existing IDocument and updated IContent as arguments? This will help the modified content to update the hashes, and we can replace the existing document with the new content. Post this, we can update other sections of the document (signature, dates etc) and create the updated document from which the stream details are captured. What the update should not allow is the change of schema. We need to run validation to ensure the new content uses the schema mentioned in the original document.